tests/hil/tests/settings/test_settings.py

```diff
@@ -207,3 +208,24 @@ async def test_cancel_all(board, device):
 
     await device.settings.set('TEST_INT', 72)
     assert None != board.wait_for_regex_in_line('Received test_int: 72', timeout_s=10)
+
+
+async def test_restart(board, device):
+    await device.settings.set('TEST_RESTART', True)
+    assert None != board.wait_for_regex_in_line('Received test_restart: true', timeout_s=10)
+    assert None != board.wait_for_regex_in_line('Ending session', timeout_s=10)
+
+    # Check that we no longer receive this settings change
+    await device.settings.set('TEST_INT', 2320)
+    with pytest.raises(RuntimeError) as e:
+        assert None != board.wait_for_regex_in_line('Received test_int', timeout_s=10)
+    assert str(e.value) == "Timeout"
+
+    # Wait for client to restart
+    assert None != board.wait_for_regex_in_line('Client restarted', timeout_s=60)
+
+    # Wait for rush of initial settings log messages to pass
+    time.sleep(2)
```

Comment:
Nit: I'd prefer `await trio.sleep(5)` instead. This won't change much in the execution, but it makes more sense to use async equivalent in async context, as well as for consistency.

Reply:
Yeah, thanks for the callout. I saw that PR you made with this and agree it's a better approach. I'll update the outstanding PRs before merging.